src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -57,9 +57,34 @@ public PersonCard(Person person, int displayedIndex) {
         address.setText(person.getAddress().value);
         email.setText(person.getEmail().value);
         description.setText(person.getDescription().value);
-        nextOfKin.setText(person.getNextOfKin().value);
+
+        // Set text for nextOfKin label if value is present, otherwise make it invisible
+        person.getNextOfKin().ifPresentOrElse(
+                nok -> {
+                    nextOfKin.setText(nok.value);
+                    showLabel(nextOfKin);
+                }, () -> hideLabel(nextOfKin)
+        );
         person.getTags().stream()
                 .sorted(Comparator.comparing(tag -> tag.tagName))
                 .forEach(tag -> tags.getChildren().add(new Label(tag.tagName)));
     }
+    /**
+     * Hides the specified label if no value is present.
+     *
+     * @param label The label to hide if no value is present.
+     */
+    private void hideLabel(Label label) {
```

Comment:
might be good to abstract to `Node` instead of `Label` if possible, because might want to use for other objects

Reply:
Is Node like the parent of Label.
<img width="652" alt="image" src="https://github.com/AY2324S2-CS2103T-W12-4/tp/assets/48405651/ff3bc48c-42b9-48d9-bf73-051fa1dce76f">
Abstracting it out gives me something like this.
Do i just import the thing?

Reply:
yeah, https://docs.oracle.com/javase/8/javafx/api/javafx/scene/Node.html